src/Framework/Framework/Controls/RadioButton.cs

```diff
@@ -78,26 +81,34 @@ protected virtual void RenderTypeAttribute(IHtmlWriter writer)
 
         protected virtual void RenderCheckedValueAttribute(IHtmlWriter writer)
         {
-            writer.AddKnockoutDataBind("checkedValue", this, CheckedValueProperty, () => {
-                var checkedValue = (CheckedValue ?? string.Empty).ToString();
-                if (!string.IsNullOrEmpty(checkedValue))
-                {
-                    writer.AddKnockoutDataBind("checkedValue", KnockoutHelper.MakeStringLiteral(checkedValue));
-                }
-            });
+            var checkedValue = GetValueOrBinding<object?>(CheckedValueProperty);
+            if (!checkedValue.ValueIsNull())
+            {
+                var checkedValueExpr = checkedValue.GetJsExpression(this);
+                writer.AddKnockoutDataBind("checkedValue", checkedValueExpr);
+            }
             RenderCheckedValueComparerAttribute(writer);
         }
 
         protected virtual void RenderCheckedAttribute(IHtmlWriter writer)
         {
             var checkedItemBinding = GetValueBinding(CheckedItemProperty);
-            if (checkedItemBinding == null)
+            if (checkedItemBinding is null)
             {
-                writer.AddKnockoutDataBind("checked", this, CheckedProperty, () => { });
-                if (!IsPropertySet(CheckedValueProperty))
+                var @checked = GetValueRaw(CheckedProperty);
+                if (@checked is IValueBinding checkedBinding)
+                {
+                    writer.AddKnockoutDataBind("checked", this, checkedBinding);
+                }
+                if (true.Equals(KnockoutHelper.TryEvaluateValueBinding(this, @checked)))
                 {
-                    throw new DotvvmControlException(this, "The 'CheckedValue' of the RadioButton control must be set. Remember that all RadioButtons with the same GroupName have to be bound to the same property in the viewmodel.");
+                    writer.AddAttribute("checked", "");
                 }
+
+                // if (!IsPropertySet(CheckedValueProperty))
```

Comment:
Please test if this would work without CheckedValue:

```
<dot:RadioButton Checked="{value: IsPerson}" GroupName="RadioGroup" Text="person" />
<dot:RadioButton Checked="{value: IsCompany}" GroupName="RadioGroup" Text="company" />

<p>Is person: {{value: IsPerson}}</p>
<p>Is company: {{value: IsCompany}}</p>
```